kernel/core/io/file.cpp

```diff
@@ -4241,10 +4300,15 @@ namespace NumeRe
                             {
                                 styleId = style->IntAttribute("numFmtId");
 
-                                // Those are time formats: 14-22, 45-47
-                                if ((styleId >= 14 && styleId <= 22) || (styleId >= 45 && styleId <= 47))
-                                    sValue = convertExcelTimeToEpoch(sValue, (styleId >= 18 && styleId <= 21)
-                                                                              || (styleId >= 45 && styleId >= 47));
+                                // Check if styleId is in vDateTimeIds or vStandardDateTimeIds set
+                                if ((vDateTimeIds.find(styleId) != vDateTimeIds.end()) || 
+                                (vStandardDateTimeIds.find(styleId) != vStandardDateTimeIds.end()))
+                                {
+
+                                    //? Is it standard time or date time? What about mixed values with date and time?
+                                    sValue = convertExcelTimeToEpoch(sValue, 
+                                    (vStandardDateTimeIds.find(styleId) != vStandardDateTimeIds.end()));

```

Comment:
Here, a possible problem will arise: the second parameter is a boolean flag for "value is only time value and does not contain date information". This is not true for the standard IDs 14-17 (as visible in the old implementation).

I recommend to have separate `std::set`s for date values and time values. That's kind of a duplicate, but yeah, it's Excel.